pkg/controllers/dnshealthcheckprobe/dnshealthcheckprobe_controller.go

```diff
@@ -77,6 +78,21 @@ func (r *DNSHealthCheckProbeReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	probeId := probeId(previous)
 
+	additionalHeaders := v1alpha1.AdditionalHeaders{}
+
+	if previous.Spec.AdditionalHeadersRef != nil {
+		secretKey := client.ObjectKey{Name: previous.Spec.AdditionalHeadersRef.Name, Namespace: previous.Namespace}
+		additionalHeadersSecret := &v1.Secret{}
+		if err := r.Client.Get(ctx, secretKey, additionalHeadersSecret); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error retrieving additional headers secret %v/%v: %w", secretKey.Namespace, secretKey.Name, err)
+		}
+		for k, v := range additionalHeadersSecret.Data {
+			additionalHeaders = append(additionalHeaders, v1alpha1.AdditionalHeader{
+				Name:  k,
```

Comment:
any validation we should do here?

Reply:
I don't see what we could realistically validate here, a little searching hasn't led me to any sort of generic header validation. Did you have anything in mind, or were there some concerns about this, that I'm missing?

Considering that these headers can only be sent in requests to clusters that have had a gateway deployed to them from the same MGC - allowing them to define any headers they like seems fine?

Reply:
well trimming space. failing on spaces in the key? the values prob not though